File: 745-FindSmallestLetterGreaterThanTarget/745-FindSmallestLetterGreaterThanTarget.java
// Last updated: 2/27/2026, 9:36:45 PM
class Solution {
    public char nextGreatestLetter(char[] letters, char target) {
        
        int start = 0;
        int n = letters.length;
        int end = n - 1;
        int mid;

        while(start <= end){
            mid = start + (end-start)/2;

            if(letters[mid] <= target) start = mid + 1;
            else end = mid - 1;
        }

        return letters[start % n];


    }
}

File: 745-FindSmallestLetterGreaterThanTarget/745-FindSmallestLetterGreaterThanTarget.py
# Last updated: 2/27/2026, 9:36:44 PM
class Solution:
    def nextGreatestLetter(self, letters: List[str], target: str) -> str:

        if letters[-1] <= target: return letters[0]

        l, h = 0, len(letters) - 1
        
        while l <= h:
            mid = (l + h )// 2
            if letters[mid] > target:
                h = mid - 1
            else:
                l = mid + 1
        
        return letters[l]